app/controllers/users/passwords_controller.rb

```diff
@@ -2,19 +2,29 @@ module Users
   class PasswordsController < Devise::PasswordsController
     include ValidEmailParameter
 
-    before_action :confirm_valid_token, only: [:edit]
-
     def create
-      RequestPasswordReset.new(params[:user][:email]).perform
+      email = params[:user][:email]
+
+      RequestPasswordReset.new(email).perform
+
+      analytics_user = User.find_by_email(email) || AnonymousUser.new
```

Comment:
Good point that nonexistent user more accurately describes the situation here. I will rename the class to `NonexistentUser`. In this case, the Null Object pattern applies because we just want to call `analytics.track_event` once and have it report the correct role without using conditionals. In this case, we always want to report a role, as opposed to doing nothing. So, defining the role for the nonexistent user is better than having a conditional like this:
```ruby
if user is nonexistent
  analytics.track_event('Reset requested for nonexistent user')
else
  analytics.track_event("Reset requested for #{user.role}")
end
```
or even uglier:
```ruby
analytics.track_event("Reset requested for #{user.present? ? user.role : 'nonexistent'}")
```